Makefile

```diff
@@ -134,17 +134,17 @@ ENVSUBST_VER := $(call get_go_version,github.com/drone/envsubst/v2)
 ENVSUBST := $(abspath $(TOOLS_BIN_DIR)/$(ENVSUBST_BIN)-$(ENVSUBST_VER))
 ENVSUBST_PKG := github.com/drone/envsubst/v2/cmd/envsubst
 
-GO_APIDIFF_VER := v0.5.0
+GO_APIDIFF_VER := v0.7.0
 GO_APIDIFF_BIN := go-apidiff
 GO_APIDIFF := $(abspath $(TOOLS_BIN_DIR)/$(GO_APIDIFF_BIN)-$(GO_APIDIFF_VER))
 GO_APIDIFF_PKG := github.com/joelanford/go-apidiff
 
-HADOLINT_VER := v2.10.0
+HADOLINT_VER := v2.12.0
 HADOLINT_FAILURE_THRESHOLD = warning
 
 SHELLCHECK_VER := v0.9.0
 
-KPROMO_VER := v3.5.1
+KPROMO_VER := v4.0.4
```

Comment:
You may need to update a line farther down in the file to `v4`; I made this mistake in CAPZ and here's my fix: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4478

Reply:
https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/8d7fcb998f1224b0fb6511ea5f127adeb1a008c0/Makefile#L150

Should end with `v4/cmd/kpromo`.

Reply:
Nice catch!